### PR TITLE
[114] adding backend code for combo box menu

### PR DESCRIPTION
**Pull request by ayya-vimala**
The backend for this works. There are issues at the frontend only.

### CODE REVIEW

File: api/db_queries.py [hunk 1]
```diff
@@ -8,13 +8,13 @@
                     FILTER segment._key == segmentnr
                     FOR segment_id IN segment.parallel_ids
                         FOR p IN parallels
+                           FILTER p._key == segment_id
```
Comment (TrebuhD) [inline api/db_queries.py]: that's a really good change 👍 It probably improved performance by a lot.

File: api/db_queries.py [hunk 2]
```diff
@@ -37,3 +37,31 @@
...
+                    filename: file.filename,
+                    category: file.category}
+"""
+
+query_items_for_filter_menu = """
+FOR category IN menu_categories
+    FILTER category.language == @language
+    SORT category.categorynr
+    FOR catfile in category.files
+        RETURN {filename: catfile}
```
Comment (TrebuhD) [inline api/db_queries.py]: We're returning an array like this:
```
[
  {
    "filename": "dn1"
  },
  {
    "filename": "dn2"
  },
  {
    "filename": "dn3"
  },
]
```
but the `filename` key seems to be unnecessary here. If we just `RETURN catfile`, we'll get a simpler array.

File: api/db_queries.py [hunk 3]
```diff
@@ -37,3 +37,31 @@
...
+"""
+
+query_items_for_category_menu = """
+FOR category IN menu_categories
+    FILTER category.language == @language
+    SORT category.categorynr
+    RETURN {category: category.category}
```
Comment (TrebuhD) [inline api/db_queries.py]: ditto, we can just return a list of categories.

File: api/main.py [hunk 4]
```diff
@@ -122,3 +122,67 @@ def root():
...
         raise HTTPException(status_code=400)
+
+
+@app.get("/menus/{language}")
+async def get_items_for_menu(language: str):
+    try:
+        db = get_db()
+        menu_query_result = db.AQLQuery(
+            query=query_items_for_menu,
+            bindVars={"language": get_language_from_languagename(language)}
```
Comment (TrebuhD) [inline api/main.py]: question: is the transforming function necessary? Can't we just pass the desired language name in the request?

File: api/db_queries.py (hunk 2 above)
Comment (ayya-vimala) [reply api/db_queries.py]: Yes, but the multi-combo-box needs an object with items with a certain name each. It is possible that way to also add more items like "displayname" that differ from the filename. Right now I only use the filename in the combo-box but maybe in the future that will change. Will have to see how it feels after a while.

File: api/db_queries.py (hunk 3 above)
Comment (ayya-vimala) [reply api/db_queries.py]: Ditto as above.

File: api/main.py (hunk 4 above)
Comment (ayya-vimala) [reply api/main.py]: Then we have to transform the language in the frontend before it is passed in. 
Or we should make the entire frontend just using the abbreviated languages and not the full language.

File: api/utils.py [hunk 5]
```diff
@@ -8,3 +8,20 @@ def get_language_from_filename(filename):
         return "chn"
     else:
         return "pli"
+
+
+def create_teststring(limitcollection,language):
```
Comment (TrebuhD) [inline api/utils.py]: get we please get a better description of this method? Having a docstring would be really nice.

File: api/utils.py [hunk 6]
```diff
@@ -8,3 +8,20 @@ def get_language_from_filename(filename):
...
+    teststring = []
+    if language == "tib" or language == "chn":
+        for item in limitcollection:
```
Comment (TrebuhD) [inline api/utils.py]: nitpicking: can we use a better variable name than `item`?

File: api/db_queries.py [hunk 7]
```diff
@@ -37,3 +37,33 @@
             RETURN { [category["category"]]: category.categoryname }
 )
 """
+
+query_items_for_menu = """
```
Comment (TrebuhD) [inline api/db_queries.py]: naming: even though this is currently used for the menu, I would avoid naming database queries after specific UI elements. IMHO a better name would describe what the query does, like `query_files_for_category` (just an example)

File: api/db_queries.py [hunk 8]
```diff
@@ -37,3 +37,33 @@
...
+        FOR file in files
+            FILTER file._key == catfile
+            SORT file.filenr
+            RETURN {displayName: file.displayName,
+                    textname: file.textname,
+                    filename: file.filename,
+                    category: file.category}
+"""
+
+query_items_for_filter_menu = """
```
Comment (TrebuhD) [inline api/db_queries.py]: ^ same as above. I think all these filter boxes technically qualify as filter menus?

File: api/db_queries.py [hunk 9]
```diff
@@ -37,3 +37,33 @@
...
+FOR category IN menu_categories
+    FILTER category.language == @language
+    SORT category.categorynr
+    FOR catfile IN category.files
+        RETURN {filename: catfile,
+                categoryname: UPPER(catfile)}
+"""
+
+query_items_for_category_menu = """
```
Comment (TrebuhD) [inline api/db_queries.py]: ditto

File: api/main.py [hunk 10]
```diff
@@ -122,3 +123,67 @@ def root():
...
+    except AQLQueryError as e:
+        print("AQLQueryError: ", e)
+        raise HTTPException(status_code=400, detail=e.errors)
+    except KeyError as e:
+        print("KeyError: ", e)
+        raise HTTPException(status_code=400)
+
+
+@app.get("/menus/filter/{language}")
+async def get_items_for_filter_menu(language: str):
```
Comment (TrebuhD) [inline api/main.py]: naming: I think the name `filter_menu` is too generic here.

**Pull request merged**